api-reference/10 UI Components/dxNumberBox/1 Configuration/format.md

```diff
@@ -8,13 +8,115 @@ default: ''
+
+- Custom format string    
+
+    Use Unicode Locale Data Markup Language (LDML) patterns to specify a custom format string. An LDML pattern consists of wildcard characters and characters displayed as is. The **format** property supports the following wildcard characters:
+
+    <table class="dx-table">
+        <tr>
+        <th>Format character</th>
+        <th>Description</th>
+        </tr>
+        <tr>
+        <td>0</td>
+        <td>A digit. Displays '0' if it is not specified in the UI.</td>
+        </tr>
+        <tr>
+        <td>&#35;</td>
+        <td>
+                A digit or nothing. One symbol represents several integer digits, but only one decimal digit.</br> 
+                For example, "#0.#" represents "123.4", but not "123.45".
+        </td>
+        </tr>
+        <tr>
+        <td>.</td>
+        <td>
+                A decimal separator. </br>
+                Displayed according to the specified locale.
+        </td>
+        </tr>
+        <tr>
+        <td>,</td>
+        <td>
+                A group separator. </br>
+                Displayed according to the specified locale.
+        </td>
+        </tr>
+        <tr>
+        <td>%</td>
+        <td>
+                The percent sign. Multiplies the input value by 100. </br>    
+                If it is enclosed in single quotes ('%'), it only adds this sign to the input value.
+        </td>
+        </tr>
+        <tr>
+        <td>;</td>
+        <td>Separates positive and negative numbers. If there is no explicit negative format, a positive number receives the "-" prefix. </td>
+        </tr>
+        <tr>
+        <td>Other characters</td>
+        <td>
+                Any character. Should be placed only at the format string's beginning or end. </br>
+                You can use the special characters above as well (in single quotation marks).
+        </td>
+        </tr>
+    </table>
+
+        <!-- tab: JavaScript -->
+        const largeNumber = 123456.789;
+
+        // Limit the decimal part to one digit
+        format: "#0.#" // 123456.7
+
+        // Add a group separator
+        format: ",##0.###" // 123,456.789
+        
+    The example below shows how to specify percentages when you work with decimals. It is necessary to add a `#` character at the start of an LDML pattern to avoid digits before `.` to be deleted.
+
+        <!-- tab: JavaScript -->
+        const smallNumber = 0.01234;
+
+        // Represent as a percentage and limit to two decimal digits
+        format: "#0.##%" // 1.23%
+
+        // Add a percent sign and limit to two decimal digits
+        format: "#0.##'%'" // 0.01%
+
+        const number = 0.80785;
+
+        // Remove all the numbers except the last digit before '.'
+        format: "0.##%" // 0.79%, would be 80.79% if the format is "#0.##%"
```

Comment:
I don't like this example - you probably would never want to format numbers like that. I suggest you remove that one. 

On the other hand, I understand that you're trying to make a point. You want to illustrate the risks of using '0' as the first character. If you really want to keep this example, then I would change its preceding comment to something like this: "Use caution if your format strings starts with a zero ('0'), because the formatted number may lose leading digits".

Reply:
I decided to leave the caution message before the code snippet and delete the last example.